File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int mn=10000;
        int pr=0;
        for(int i=0;i<n;i++){
            if(prices[i]<mn){
                mn=prices[i];
            }
            else{
                pr=pr+prices[i]-mn;
                mn=prices[i];
            }
        }
        return pr;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int pr=0,n=prices.size(),mx=0;
        for(int i=n-1;i>=0;i--){
            if(mx<prices[i]){
                mx=prices[i];
            }
            else{
                pr=max(pr,mx-prices[i]);
            }
        }
        return pr;
    }
};
